.github/workflows/tests-outerloop.yml

```diff
@@ -12,88 +12,60 @@ concurrency:
 
 jobs:
 
-  test:
-    name: ${{ matrix.os.title }}
-    runs-on: ${{ matrix.os.name }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os:
-          - name: ubuntu-latest
-            title: Linux
-          - name: windows-latest
-            title: Windows
+  generate_tests_matrix:
+    name: Generate test runsheet
+    runs-on: windows-latest
+    if: ${{ github.repository_owner == 'dotnet' }}
+    outputs:
+      runsheet: ${{ steps.generate_tests_matrix.outputs.runsheet }}
     steps:
-      - name: Setup vars (Linux)
-        if: ${{ matrix.os.name == 'ubuntu-latest' }}
-        run: |
-          echo "DOTNET_SCRIPT=./dotnet.sh" >> $GITHUB_ENV
-          echo "BUILD_SCRIPT=./build.sh" >> $GITHUB_ENV
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
-      - name: Setup vars (Windows)
-        if: ${{ matrix.os.name == 'windows-latest' }}
+      # We need to build the whole solution, so that we can interrogate each test project
+      # and find out whether it contains any quarantined tests.
+      - name: Build the solution
         run: |
-          echo "DOTNET_SCRIPT=.\dotnet.cmd" >> $env:GITHUB_ENV
-          echo "BUILD_SCRIPT=.\build.cmd" >> $env:GITHUB_ENV
+          ./build.cmd -restore -build -c Release -ci /p:CI=false /p:GeneratePackageOnBuild=false /p:InstallBrowsersForPlaywright=false /bl:./artifacts/log/Release/build.binlog
```

Comment:
Note: this binlog is quite sizable but useful for investigations. We should remove it in some not too distant future, however.